extensions/parquet/compression/src/main/java/io/deephaven/parquet/compress/DeephavenCodecFactory.java

```diff
@@ -0,0 +1,141 @@
+package io.deephaven.parquet.compress;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.compress.CodecPool;
+import org.apache.hadoop.io.compress.CompressionCodec;
+import org.apache.hadoop.io.compress.CompressionCodecFactory;
+import org.apache.hadoop.io.compress.CompressionInputStream;
+import org.apache.hadoop.io.compress.Decompressor;
+import org.apache.parquet.bytes.BytesInput;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Deephaven flavor of the Hadoop/Parquet CompressionCodec factory, offering support for picking codecs from
+ * configuration or from the classpath (via service loaders), while still offering the ability to get a
+ * CompressionCodecName enum value having loaded the codec in this way.
+ */
+public class DeephavenCodecFactory {
+
+    private static final List<Class<?>> CODECS = io.deephaven.configuration.Configuration.getInstance()
+            .getStringSetFromPropertyWithDefault("DeephavenCodecFactory.codecs", Collections.emptySet()).stream()
+            .map((String className) -> {
+                try {
+                    return Class.forName(className);
+                } catch (ClassNotFoundException e) {
+                    throw new IllegalStateException("Can't find codec with name " + className);
+                }
+            }).collect(Collectors.toList());
+
+    private static volatile DeephavenCodecFactory INSTANCE = new DeephavenCodecFactory(CODECS);
```

Comment:
This initialization seems incorrect. Don't you want it to start out `null` so it can be instantiated lazily or supplied with `setInstance`?
```suggestion
    private static volatile DeephavenCodecFactory INSTANCE;
```